tcms/xmlrpc/tests/test_testrun.py

```diff
@@ -202,3 +203,64 @@ def test_create_with_product_version(self):
         # the result is still using product_version from TR.plan.product_version
         # not the one we specified above
         self.assertEqual(result['product_version'], self.plan.product_version.value)
+
+
+class TestUpdateTestRun(XmlrpcAPIBaseTest):
+    def _fixture_setup(self):
+        super()._fixture_setup()
+
+        self.test_run = TestRunFactory()
+
+        self.updated_test_plan = TestPlanFactory()
+        self.updated_build = BuildFactory()
+        self.updated_summary = 'Updated summary.'
+        self.updated_stop_date = '2020-05-05'
+
+    def test_successful_update(self):
+        update_fields = {
+            'plan': self.updated_test_plan.pk,
+            'build': self.updated_build.pk,
+            'summary': self.updated_summary,
+            'stop_date': self.updated_stop_date
+        }
+
+        # assert test plan is not already updated
+        self.assertNotEqual(self.updated_test_plan, self.test_run.plan.name)
+        self.assertNotEqual(self.updated_build, self.test_run.build.name)
+        self.assertNotEqual(self.updated_summary, self.test_run.summary)
+        self.assertNotEqual(self.updated_stop_date, str(self.test_run.stop_date))
+
+        updated_test_run = self.rpc_client.exec.TestRun.update(self.test_run.pk, update_fields)
+        self.test_run.refresh_from_db()
+
+        # compare result, returned form API call with test run from DB
+        self.assertEqual(updated_test_run['plan'], self.test_run.plan.name)
+        self.assertEqual(updated_test_run['build'], self.test_run.build.name)
+        self.assertEqual(updated_test_run['summary'], self.test_run.summary)
+        self.assertEqual(updated_test_run['stop_date'], str(self.test_run.stop_date))
+
+        # compare result, returned form API call with params sent to the API
+        self.assertEqual(updated_test_run['plan'], self.updated_test_plan.name)
+        self.assertEqual(updated_test_run['build'], self.updated_build.name)
+        self.assertEqual(updated_test_run['summary'], self.updated_summary)
+        self.assertEqual(
+            updated_test_run['stop_date'],
+            str(datetime.strptime(self.updated_stop_date, '%Y-%m-%d'))
+        )
+
+    def test_wrong_date_format(self):
+        update_fields = {
+            'stop_date': '10-10-2010'
+        }
+
+        test_run_pre_update = self.test_run
```

Comment:
You don't need this intermediate variable.

Reply:
Then how do I assert things have not changed? What do I compare `self.test_run` with?

Reply:
The only way to do this is to somehow keep reference to the old values as scalars. For the purposes of the test you can keep a reference to the PKs or names/values of all fields you are trying to update then assert against them. 

Another possibility is to use `assertNotEqual` against values from the `update_fields` dict. You can extend the dict with a few more fields if you go this route.

See also comment below about the last group of assert statements.

Reply:
I decided to go with the second option and extend the `update_params` object. 

However, now the final assert (about the date) looks like this:
```python
self.assertNotEqual(datetime.strptime(update_fields['stop_date'], '%d-%m-%Y'),self.test_run.stop_date)
```
This is so, because `update_fields['stop_date']` is a `string` and `test_run.stop_date` is a `datetime`. 
if I compared - `update_fields['stop_date']` to `str(test_run.stop_date)`, then `str(test_run.stop_date)` will look something like this `2010-10-10 00:00`. So even if we did have a bug, and this API call succeeds and updates the date, this test will still pass, because of the type mismatch in the assertions. 
This is why I decided to go with this implementation.